src/particle/corsika/corsika7id.py

```diff
@@ -0,0 +1,217 @@
+        if 200 <= cid < 5699:
+            return cls(cid), ismother
+
+        if cid in _bimap:
+            return cls(cid), ismother
+
+        raise MatchingIDNotFound(
+            f"Non-existent Corsika7ID for particle description {particle_description}!"
+        )
+
+    @classmethod
+    def _is_non_particle_id(cls: type[Self], corsikaid: int) -> bool:
+        """
+        Returns True if the ID is valid but does not correspond to a particle, False otherwise.
+        """
+        return (
+            corsikaid in _non_particles
+            or corsikaid // 1000 == 8888
+            or corsikaid == 9900
+        )
+
+    def is_particle(self) -> bool:
+        """
+        Returns True if the ID really belongs to a particle, since some are for example additional information.
+
+        Examples
+        --------
+        >>> mu_minux = Corsika7ID(6)
+        >>> mu_minus.is_particle()
+        True
+        >>> mu_info = Corsika7ID(76)
+        >>> mu_info.is_particle()
+        False
+        >>> mu_info.name()
+        'μ− add. info.'
+        """
+        iid = int(self)
+
+        return not self._is_non_particle_id(iid)
+
+    def name(self) -> str:
+        """
+        Returns a human readable name of the Corsika7ID.
+        This also works for non-particles (is_particle()==false).
+
+        Raises
+        ------
+        ParticleNotFound
+            If it is a 'valid' PDG particle, but unknown.
+            This for example happens with strange nuclei, which are not in the nuclei list.
+
+        Examples
+        --------
+        >>> mu_minus = Corsika7ID(6)
+        >>> mu_minus.is_particle()
+        True
+        >>> mu_minus.name() # For a particle, this returns the same name as `Particle.name`
+        'mu'
+        >>> mu_info = Corsika7ID(76)
+        >>> mu_info.is_particle()
+        False
+        >>> mu_info.name()
+        'μ− add. info.'
+        >>> ch_photons_of = Corsika7ID(9900)
+        >>> ch_photons_of.is_particle()
+        False
+        >>> ch_photons_of.name()
+        'Cherenkov photons on particle output file'
+        """
+        from ..particle.particle import Particle
+
+        if self.is_particle():
+            return str(Particle.from_pdgid(self.to_pdgid()).name)
+
+        iid = int(self)
+
+        if iid in _non_particles:
+            return _non_particles[iid]
+
+        if iid // 1000 == 8888:
+            return "weights of preceding particle (MULTITHIN option)"
+
+        if iid == 9900:
+            return "Cherenkov photons on particle output file"
+
+        raise RuntimeError("This should be unreachable.")
+
+    def to_pdgid(self) -> PDGID:
+        """
+        Raises
+        ------
+        InvalidParticle
+            If it is a valid Corsika particle, but not a valid PDGID.
+
+        Examples
+        --------
+        >>> Corsika7ID(6).to_pdgid()
+        <PDGID: 13>
+        >>> Corsika7ID(76).to_pdgid()
+        ...
```

Comment:
Are these 3 dots really relevant?

As far as the sentence "InvalidParticle: The Corsika7ID <Corsika7ID: 76> is not a valid PDGID" is concerned, is this the right message to convey? Is it not that for example 76 is a Corsika7 specific ID with no particle equivalent as provided by the is_particle method? MAybe better to rephrase in the sentence and this example. WDYT?

Reply:
I totaly agree!